src/components/PostsList.tsx

```diff
@@ -1,85 +1,99 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+import { apiActions } from '../utils/apiAction';
+import { User } from '../types/User';
+import { ErrorNotification } from '../types/ErrorNotification';
+import { Loader } from './Loader';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+type Props = {
+  setPosts: (posts: Post[]) => void;
+  userSelected: User;
+  setErrorNotification: (error: (prevError: ErrorNotification)
+  => ErrorNotification) => void;
+  posts: Post[];
+  setSelectedPost: React.Dispatch<React.SetStateAction<Post | null>>;
+  selectedPost: Post | null;
+  isPostsLoading: boolean;
+  setIsPostsLoading: (postL: boolean) => void;
+};
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<Props> = ({
+  setPosts,
+  userSelected,
+  setErrorNotification,
+  posts,
+  setSelectedPost,
+  selectedPost,
+  isPostsLoading,
+  setIsPostsLoading,
+}) => {
+  useEffect(() => {
+    setIsPostsLoading(false);
 
-      <tbody>
-        <tr data-cy="Post">
-          <td data-cy="PostId">17</td>
+    apiActions.getUserPosts(userSelected.id)
+      .then(setPosts)
+      .catch(() => {
+        setErrorNotification((errorNotification) => ({
+          ...errorNotification,
+          posts: true,
+        }));
+      })
+      .finally(() => setIsPostsLoading(true));
+  }, [userSelected.id]);
```

Comment:
Same here
![2023-10-18_15h50_57](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/61904995/4c62b232-9129-4198-929a-1be897bcf20f)